sdk/storage/Azure.Storage.Files.Shares/api/Azure.Storage.Files.Shares.netstandard2.0.cs

```diff
@@ -792,6 +800,7 @@ public partial class ShareProperties
         public System.DateTimeOffset? AccessTierChangeTime { get { throw null; } }
         public string AccessTierTransitionState { get { throw null; } }
         public System.DateTimeOffset? DeletedOn { get { throw null; } }
+        public string EnabledProtocols { get { throw null; } }
```

Comment:
Note that `EnabledProtocols` is a string, but `RootSquash` is an enum.  I did this because it is not possible to create a Flags enum in swagger, and in the future, it will be possible enable multiple protocols on a share.  I manually created a Flags Enum `ShareEnabledProtocols` to be used in `ShareCreateOptions`.  To use the flag enum here, we would have to make the original generated `ShareProperties` class internal, and hand-write a new one and the conversion between.  This would also break swagger for the other platforms.  We can do this, but it would be more work.

Reply:
I think use `string` in generated code is fine and we should do some wrapping in the convenience layer.
Note that in the future, the possible values for the `EnabledProtocols` could be 
```
"SMB;NFS"
"SMBv2;SMBv3;NFSv4"
```
"SMB"/"NFS" without version is to support all versions. And "SMBv2;SMBv3;" is to support both SMB v2 and v3.

To me, this is similar to the `include` option of List Blob except that the service is accepting a joined string instead of an array.

Reply:
@ljian3377, we are going to leave this as a string in the generated code, but we are going to make it a flags enum in the ShareItem that is actually returned to the customer.